### PR TITLE
impl(otel): bazel loads OpenTelemetry

### PR DESCRIPTION
**Pull request by dbolduc**
Part of the work for #10283

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10294)

### CODE REVIEW

File: WORKSPACE.bazel [hunk 1]
```diff
@@ -34,3 +34,7 @@ grpc_deps()
 load("@com_github_grpc_grpc//bazel:grpc_extra_deps.bzl", "grpc_extra_deps")
 
 grpc_extra_deps()
+
+load("@io_opentelemetry_cpp//bazel:repository.bzl", "opentelemetry_cpp_deps")
```
Comment (coryan) [inline WORKSPACE.bazel:38]: Will we need to change all the quickstart WORKSPACE files too?  Not now, but it a future PR?
Maybe it is time to create the multiple workspace functions like TensorFlow does.

File: bazel/google_cloud_cpp_deps.bzl [hunk 2]
```diff
@@ -186,3 +186,14 @@ def google_cloud_cpp_deps():
             patch_args = ["-p1"],
             patches = [Label("//bazel:configure_template.bzl.patch")],
         )
+
+    # Open Telemetry
+    if "io_opentelemetry_cpp" not in native.existing_rules():
```
Comment (coryan) [inline bazel/google_cloud_cpp_deps.bzl:191]: The renovate configuration will also need a PR, one where both CMake and Bazel get changed in a single PR.

**Review by coryan — APPROVED** [review]

File: bazel/google_cloud_cpp_deps.bzl (hunk 2 above)
Comment (dbolduc) [reply bazel/google_cloud_cpp_deps.bzl:191]: Ack. I added this as an extra task to the GH issue.

File: WORKSPACE.bazel (hunk 1 above)
Comment (dbolduc) [reply WORKSPACE.bazel:38]: > Will we need to change all the quickstart WORKSPACE files too? Not now, but it a future PR?

Yeah, we will. That is a task for Milestone 2.

> Maybe it is time to create the multiple workspace functions like TensorFlow does.

:shrug:

**Pull request merged**